bots/image-prepare

```diff
@@ -46,100 +47,141 @@ def main():
-    elif not build_image:
-        build_image = get_build_image(install_image or args.image)
+    # Depending on the operating system ignore some things
+    skips = [ "cockpit-integration-tests" ]
+    if args.address:
+        skips.append("cockpit-tests")
+        args.install_only = True
+
+    if args.install_only and args.build_only:
+        sys.stderr.write("image-prepare: use of --install-only with --build-only is incompatible\n")
+        return 2
 
     # Default to putting build output in bots directory
     if "TEST_ATTACHMENTS" in os.environ:
         results = os.path.join(os.environ["TEST_ATTACHMENTS"], "build-results")
     else:
         results = os.path.join(BOTS, "build-results")
 
+    # Make sure any images are downloaded
     try:
-        if build_image:
+        if not args.address and not os.path.exists(base_image):
+            subprocess.check_call([ "image-download", base_image])
+        if build_image and not os.path.exists(build_image):
             subprocess.check_call([ "image-download", build_image])
-        if install_image:
-            subprocess.check_call([ "image-download", install_image])
     except OSError, ex:
         if ex.errno != errno.ENOENT:
             raise
         sys.stderr.write("image-prepare: missing tools to download images\n")
     except subprocess.CalledProcessError as e:
-        print "unable to download all necessary images", e
+        sys.stderr.write("image-prepare: unable to download all necessary images\n")
         return 1
 
     try:
-        build_and_install(install_image, build_image, results, args={
-            "verbose": args.verbose,
-            "sit": args.sit,
-            "quick": args.quick,
-            "build_image": build_image,
-            "build_only": args.build_only,
-            "install_only": args.install_only,
-            "containers": args.containers,
-            "address": args.address,
-        })
-    except RuntimeError as ex:
+        if not args.install_only:
+            build_and_install(build_image, results, skips, args)
+        if not args.build_only and build_image != base_image:
+            only_install(base_image, results, skips, args, args.address)
+    except (RuntimeError, testvm.Failure) as ex:
         sys.stderr.write("image-prepare: {0}\n".format(str(ex)))
         return 2
 
+    # The remaining images not yet prepared are symlinked from the bots/images
+    # directory into the test/images directory. This allows for other branches to
+    # run without modification, and allows tests to load images that they didn't
+    # install or need to install cockpit on (above).
+    link_remaining_images()
+
     return 0
 
+def link_remaining_images():
+    bots_images = os.path.join(BOTS, "images")
+    test_images = os.path.join(TEST, "images")
+    if not os.path.exists(test_images):
+        os.makedirs(test_images)
+    for image in os.listdir(bots_images):
+        if "." in image:
+            continue
```

Comment:
That looks wrong and is most likely a real bug -- for example, we have a rhel-7.4 image now. `os.listdir()` already filters out `.` and `..`. If you are concerned about hidden files, please use `image.startswith('.')`, but I figure we don't really need that.